### PR TITLE
Adding docs about the build section

### PR DESCRIPTION
**Pull request by gbenhaim**
closes https://github.com/lago-project/lago/issues/509

Signed-off-by: gbenhaim <galbh2@gmail.com>

### CODE REVIEW

File: docs/BUILD.rst [hunk 1]
```diff
@@ -0,0 +1,86 @@
+##########
+Lago build
+##########
+
+Lago allows to build / configure VM disks during init stage.
+In the init file, the key ``build`` should be added to each disk that needs to be configured.
```
Comment (nvgoldin) [inline docs/BUILD.rst]: I would add here a small note/warning that this happens behind-the-back of the OS and  before the VM starts, thus should be used with care.

File: docs/BUILD.rst [hunk 2]
```diff
@@ -0,0 +1,86 @@
...
+
+`virt-customize` can be called only on disks which contains an OS.
+
+Depends on the version of `virt-customize` installed on your system (it can vary between
+different OS), all the valid options for `virt-customize` can be specified in the init file.
+For the full list of options please refer to `virt-customize documentation`_.
+
+There is a special case when using `virt-customize` to inject ssh keys. If the
+empty string is provided to ``ssh-inject`` option, Lago will replace it with
+the path to the environment's ssh key.
```
Comment (nvgoldin) [inline docs/BUILD.rst]: maybe /to the environment's ssh key/lago's self generated ssh keys/g to make it obvious it is not the user's own ssh keys.

File: docs/LagoInitFile.rst [hunk 3]
```diff
@@ -114,23 +114,31 @@ domains
...
                       build:
-                          - virt-customize
+                          - virt-customize:
                                 ssh-inject: ''
+                                touch: [/root/file1, /root/file2]
 
-                Which will ensure Lago's generated SSH keys will be injected
-                into the VM. This is useful when you don't want to run the
-                bootstrap stage.
+                See `BUILD`_ section for details.
```
Comment (nvgoldin) [inline docs/LagoInitFile.rst]: doesn't have to be upper case

File: docs/index.rst [hunk 4]
```diff
@@ -24,6 +24,7 @@ Getting started
     Templates
     Configuration
     CPU
+    BUILD
```
Comment (nvgoldin) [inline docs/index.rst]: I think it can go before CPU - more important and commonly used.

**Comment by nvgoldin** [comment]
well written :)
just few small comments

File: docs/index.rst (hunk 4 above)
Comment (gbenhaim) [inline docs/index.rst]: done

File: docs/LagoInitFile.rst (hunk 3 above)
Comment (gbenhaim) [inline docs/LagoInitFile.rst]: done

File: docs/BUILD.rst (hunk 2 above)
Comment (gbenhaim) [inline docs/BUILD.rst]: done

File: docs/BUILD.rst (hunk 1 above)
Comment (gbenhaim) [inline docs/BUILD.rst]: done

**Comment by gbenhaim** [comment]
ci test please

**Comment by gbenhaim** [comment]
ci merge please

**Pull request merged**